util/src/main/java/com/google/cloud/hadoop/util/authorization/StorageRequestAuthorizer.java

```diff
@@ -0,0 +1,154 @@
+package com.google.cloud.hadoop.util.authorization;
+
+import com.google.api.client.http.json.JsonHttpContent;
+import com.google.api.services.storage.Storage;
+import com.google.api.services.storage.StorageRequest;
+import com.google.api.services.storage.model.ComposeRequest;
+import com.google.api.services.storage.model.StorageObject;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.flogger.GoogleLogger;
+import java.net.URI;
+import java.nio.file.AccessDeniedException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/** Authorize storage requests using user specified authorization handler. */
+public class StorageRequestAuthorizer {
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+
+  private AuthorizationHandler authorizationHandler = null;
+
+  /**
+   * Create and init AuthorizationHandler from Class definition.
+   *
+   * @param handlerClass Class object of Authorization handler.
+   * @param properties String to String map of properties.
+   */
+  public StorageRequestAuthorizer(
+      Class<? extends AuthorizationHandler> handlerClass, Map<String, String> properties)
+      throws ReflectiveOperationException {
+    // Check if it is set by user
+    if (handlerClass != null) {
```

Comment:
Better to check if passed parameter is not null and do not create this object if `handlerClass` is null:

```java
checkNotNull(handlerClass, "handlerClass can not be null");
```